File: backend/internal/models/chat_history.go
package models

import (
	"github.com/google/uuid"
)

type ChatHistory struct {
	ID                uuid.UUID `json:"id"`
	ChatID            uuid.UUID `json:"chat_id"`
	UserID            uint      `json:"user_id"`
	MessageText       string    `json:"message_text"`
	Type              string    `json:"type"`
	Emotion           *int      `json:"emotion"`
	EmotionLabel      *string   `json:"emotion_label"`
	EmotionConfidence *float64  `json:"emotion_confidence"`
	CreatedAt         int64     `json:"created_at"`
}

File: ai-diary/ai-diary/Models/ChatHistory.swift
import Foundation

struct ChatHistory: Identifiable, Codable {
    let id: String
    let chatId: String
    let userId: String
    let messageText: String
    let type: String
    let emotion: Int?
    let emotionLabel: String?
    let emotionConfidence: Double?
    let createdAt: Date
}

